spec/models/user_spec.rb

```diff
@@ -173,6 +186,23 @@
         expect(existing_user.login).to eq("new")
       end
     end
+
+    context "as an admin" do
+      let(:admin) { create(:admin) }
+
+      before do
+        User.current_user = admin
+        existing_user.update!(email: "new_email@example.com", profile: nil)
```

Comment:
Can you explain why `profile: nil` is necessary here?

Reply:
It's a workaround for the requirement that a ticket be associated with this admin change (otherwise updating fails with a validation error). I could do some mocking here similar to https://github.com/otwcode/otwarchive/blob/master/spec/models/concerns/justifiable_spec.rb if you'd like, but I didn't want to add too much that's not directly related to the test

Reply:
Hmm, I'm not sure that's what the problem is. If I reload the user or even just use `let!` to create it before trying to update it, the tests pass without `profile: nil`. I also don't have that problem when trying to update the user from the console, even if I have `User.current_user` set to an admin.

My _guess_ is it relates to `before_create :create_default_associateds` not firing for some reason -- it's responsible for creating the user profile, and the error I get is that the profile doesn't exist, making the user invalid, not that I'm missing a ticket number... but if that's the case, I'm not sure why the other specs don't have that issue.

@redsummernight or @tickinginstant, do you know what might be up here? I feel like I've run into something along this line before and one of you has set me straight, but I am drawing a blank...

Reply:
Using `let` for `existing_user` means that the user object is created the first time that `existing_user` is accessed. In this test case (but not any of the others), that occurs after setting `User.current_user = admin`, which means that the system is treating it like someone logged in as an `Admin` is trying to create a `User` account. That's causing the `Justifiable` callbacks to trigger (because admins aren't supposed to be modifying user profiles willy-nilly), which prevents the profile from being saved alongside the user.

Switching to `let!` seems like a reasonable solution to me, since the test failure here effectively comes from the fact that `existing_user` doesn't exist before the admin "logs in" to try to update it.

Reply:
Thank you both for the explanations! I've updated the tests to use `let!` so now the weird `nil` update isn't needed anymore